### PR TITLE
feat: Add types file for sdk

### PR DESCRIPTION
**Pull request by nishuzumi**
Add types file for sdk

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,5 +35,6 @@
     "@rollup/plugin-node-resolve": "^15.2.3",
     "rollup": "^4.14.1",
     "rollup-plugin-terser": "^7.0.2"
-  }
+  },
+  "types": "index.d.ts"
```
Comment (iteyelmp) [inline package.json:39]: Would it be better to package the type files together into the dist directory?  There may be multiple entry files in the future.

Comment (nishuzumi) [reply package.json:39]: In theory, this type file should only be used for retrieval, even if it is not placed in dist, it should be automatically read by idea, or do you have any good changes?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
+declare module 'ethstorage-sdk' {
+    import { ethers } from 'ethers/lib.esm';
```
[the remark this answers is not in the document]
Comment (nishuzumi) [reply index.d.ts]: chang it with 
```
import {ethers} from 'ethers'
```

File: package.json (hunk 1 above)
Comment (nishuzumi) [reply package.json:48]: How about
```
  "types": "dist/index.d.ts",
  "exports": {
    ".": {
      "require": "./dist/index.cjs.js",
      "import": "./dist/index.esm.js",
      "types": "./types/index.d.ts"
    },
    "./file": {
      "require": "./dist/file.cjs.js",
    }
  },
```

Comment (nishuzumi) [reply package.json:48]: @iteyelmp What do you think?

Comment (iteyelmp) [reply package.json:48]: Looks okay

**Comment by nishuzumi** [comment]
Solved @iteyelmp

**Review by iteyelmp — APPROVED** [review]

**Pull request merged**